versioned/storage/telemetry/src/main/java/org/projectnessie/versioned/storage/telemetry/OpenTelemetryTraced.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.telemetry;
+
+import static org.projectnessie.versioned.storage.telemetry.Traced.tagName;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+
+final class OpenTelemetryTraced implements Traced {
+
+  private final Span span;
+  private final Scope scope;
+
+  @SuppressWarnings("MustBeClosedChecker")
+  OpenTelemetryTraced(Tracer tracer, String spanName) {
+    span = tracer.spanBuilder(spanName).startSpan();
+    scope = span.makeCurrent();
```

Comment:
I wonder if Quarkus is able to automagically instrument the application at build time based on `@WithSpan` annotations (I did not read the docs :) ... yet)

Reply:
But... this module does not depend on Quarkus

Reply:
Missed that :facepalm: We can keep this with custom instrumentation and revisit later. Perhaps there's a way to run standard OTel instrumentors as a build step (they can certainly instrument in runtime, but I think build-time is preferable for Nessie in this case).